lib/rspec/mocks/message_expectation.rb

```diff
@@ -683,6 +721,22 @@ def call(*_args_to_ignore, &_block)
       end
     end
 
+    # Handles the implementation of an `and_invoke` implementation.
+    # @private
+    class AndInvokeImplementation
+      def initialize(procs_to_invoke)
+        @procs_to_invoke = procs_to_invoke
+      end
+
+      def call(*_args_to_ignore, &_block)
+        if @procs_to_invoke.size > 1
+          @procs_to_invoke.shift.call
+        else
+          @procs_to_invoke.first.call
+        end
```

Comment:
This has a bug if there are nils or an empty argument splat is used.

Reply:
Good catch -- I'll validate the inputs immediately and raise some error (will try to find prior art).

Reply:
Should be fixed in the last couple commits. :+1: